### PR TITLE
Add desktop-notifier module

### LINKED ISSUE #41: implement linux desktop notifications
Make it possible to send Pulsar threat alerts as notifications when running on a Linux desktop.

Support for principal desktop environments: gnome, plasma kde, etc.

### PR DESCRIPTION
**Pull request by MatteoNardi**
# Pull Request Title

Fix #41 by creating a new [desktop-notifier module](https://github.com/Exein-io/pulsar/tree/desktop-notifier/modules/desktop-notifier)

image

## I have 

- [x] run `cargo fmt`;
- [x] run `cargo clippy`;
- [x] run `cargo test`and all tests pass;
- [x] linked to the originating issue (if applicable).

### CODE REVIEW

File: modules/desktop-notifier/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+license = "Apache-2.0"
+edition = "2021"
+
+[dependencies]
+pulsar-core = { path = "../../pulsar-core" }
+bpf-common = { path = "../../bpf-common" }
+
+tokio = { version = "1", features = ["full"] }
+log = "0.4"
+notify-rust = "4.7.0"
```
Comment (banditopazzo) [inline modules/desktop-notifier/Cargo.toml:13]: notify-rust seems not used

File: modules/desktop-notifier/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            msg = receiver.recv() => msg?,
+        };
+        if let Some(Threat { source, info }) = &event.header().threat {
+            let payload = &event.payload();
+            notify(
+                &config,
+                vec![
+                    format!("Pulsar module {source} identified a threat"),
+                    format!("{info}\n Source event: {payload}"),
+                ],
```
Comment (banditopazzo) [inline modules/desktop-notifier/src/lib.rs:45]: better to pass data and then create title and content inside the function, but it's only a matter of style

**Review by banditopazzo — CHANGES_REQUESTED** [review]

File: modules/desktop-notifier/Cargo.toml (hunk 1 above)
Comment (MatteoNardi) [reply modules/desktop-notifier/Cargo.toml:13]: ops! thanks

File: modules/desktop-notifier/src/lib.rs (hunk 2 above)
Comment (MatteoNardi) [reply modules/desktop-notifier/src/lib.rs]: I wanted a low level function which could be used also for other notifications, but I agree formatting the title and content in the main task function feels kind of weird. I've created an intermediate `handle_event` function which calls the low-level one.

**Pull request merged**